librecad/src/main/qc_mdiwindow.h

```diff
@@ -115,7 +115,7 @@ public slots:
     bool has_children();
 
 signals:
-    void signalClosing(QC_MDIWindow*);
+    bool signalClosing(QC_MDIWindow*);
```

Comment:
[Qt documentation](https://doc.qt.io/qt-5/signalsandslots.html#signals) is very clear about signals:
```
They can never have return types (i.e. use void).
```
I know that moc and compiler accept this and that it basically works under certain conditions.
But for LibreCAD, which is build by many people, with different operating systems and a bunch of different compilers and Qt versions we should avoid this.

I have to take a more detailed look to understand the issue and your solution to possibly find a more robust and compatible solution.

Reply:
Ok you right. There some other possible solution: 

1. Refactor the handling of closing a MDI window; ask the user in MDI and if he still want close the window emit a Save signal to the Application instance
2. add a second parameter to signal and  slotFileClosing of type QCloseEvent pointer and send it to Application instance, so it can handled it correct there.